include/task.h

```diff
@@ -144,11 +149,12 @@ namespace detail {
 		virtual ~fence_promise() = default;
 
 		virtual void fulfill() = 0;
+		virtual allocation_id get_user_allocation_id() = 0;
 	};
 
 	struct task_geometry {
 		int dimensions = 0;
-		range<3> global_size{0, 0, 0};
+		range<3> global_size{1, 1, 1};
```

Comment:
The IDAG unconditionally throws its work assignment routine `split_task_execution_range` at every `execution_command`. This causes `split_1d` / `split_2d` to divide by zero if the global size is zero, which happens because `master_node_host_task`s default-construct their `task_geometry`. We disallow empty global ranges in kernel launches / distributed host tasks anyway, so I believe that making the geometry unit-sized by default is sound.

Reply:
That makes sense. I do feel like `master_node_host_tasks` shouldn't default-construct their task geometry in the first (rather have something like `task_geometry::unity`), but that's not related to this PR.